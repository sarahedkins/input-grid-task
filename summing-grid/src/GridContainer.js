import React, { Component } from 'react';
import { NumberInput } from './NumberInput';

const getSum = (...params) => {
  let result = 0;
  params.forEach(param => {
    const number = parseInt(param, 10);
    if (!isNaN(number)) {
      result += number;
    }
  });
  return result;
}

class GridContainer extends Component {
  constructor() {
    super();
    this.state = {
      names: ["numberOne", "numberTwo", "numberThree"],
      numberOne: 0,
      numberTwo: 0,
      numberThree: 0,
    };
  }

  handleNumberChange = event => {
    this.setState({
      [event.target.name]: event.target.value,
    });
  }

  render() {
    const { numberOne, numberTwo, numberThree } = this.state;
    return (
      <div className="grid-border flex flex-wrap justify-content">
        {this.state.names.map(name => (
          <NumberInput
            key={name}
            name={name}
            onChange={this.handleNumberChange}
          />
        ))}
        <div className="grid-border flex-item">
          {getSum(numberOne, numberTwo, numberThree)}
        </div>
      </div>
    );
  }
}

export default GridContainer;
