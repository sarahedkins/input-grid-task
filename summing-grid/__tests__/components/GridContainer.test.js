import React from 'react';
import { shallow } from 'enzyme';
import GridContainer from '../../src/components/GridContainer';

describe('GridContainer', () => {
  it('should render', () => {
    const wrapper = shallow(<GridContainer />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle number changes', () => {
    const wrapper = shallow(<GridContainer />);
    const event = {
      target: {
        name: "numberOne",
        value: "500",
      }
    };
    wrapper.instance().handleNumberChange(event);
    expect(wrapper.state().numberOne).toEqual("500");
  });
});
