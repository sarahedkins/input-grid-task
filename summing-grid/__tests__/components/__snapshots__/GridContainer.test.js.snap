// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridContainer should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GridContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <NumberInput
            name="numberOne"
            onChange={[Function]}
          />,
          <NumberInput
            name="numberTwo"
            onChange={[Function]}
          />,
          <NumberInput
            name="numberThree"
            onChange={[Function]}
          />,
        ],
        <div
          className="grid-border flex-item"
        >
          0
        </div>,
      ],
      "className": "grid-border flex flex-wrap justify-content",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "numberOne",
        "nodeType": "function",
        "props": Object {
          "name": "numberOne",
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "numberTwo",
        "nodeType": "function",
        "props": Object {
          "name": "numberTwo",
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "numberThree",
        "nodeType": "function",
        "props": Object {
          "name": "numberThree",
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0,
          "className": "grid-border flex-item",
        },
        "ref": null,
        "rendered": 0,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <NumberInput
              name="numberOne"
              onChange={[Function]}
            />,
            <NumberInput
              name="numberTwo"
              onChange={[Function]}
            />,
            <NumberInput
              name="numberThree"
              onChange={[Function]}
            />,
          ],
          <div
            className="grid-border flex-item"
          >
            0
          </div>,
        ],
        "className": "grid-border flex flex-wrap justify-content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "numberOne",
          "nodeType": "function",
          "props": Object {
            "name": "numberOne",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "numberTwo",
          "nodeType": "function",
          "props": Object {
            "name": "numberTwo",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "numberThree",
          "nodeType": "function",
          "props": Object {
            "name": "numberThree",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0,
            "className": "grid-border flex-item",
          },
          "ref": null,
          "rendered": 0,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
